libraries/lib-numeric-formats/NumericConverterRegistry.h

```diff
@@ -0,0 +1,117 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/**********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file NumericConverterRegistry.h
+
+ Dmitry Vedenko
+
+ **********************************************************************/
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "Registry.h"
+#include "NumericConverterType.h"
+
+struct NumericConverterFormatter;
+
+struct NUMERIC_FORMATS_API NumericConverterConfig final
+{
+   double sampleRate { 44100.0 };
+
+   double tempo { 120.0 };
+   int upperTimeSignature { 4 };
+   int lowerTimeSignature { 4 };
```

Comment:
This is the stuff that the factory type below needs: the parameters of a formatter that are not known at compile time.

And I see beat and bar specific stuff here, and I think again that this is a failure of open endedness.

Maybe the argument of the factory should just be `void*` after all?  Or, `std::any` so that there could be some dynamic type checking by the function, that the expected sort of data are given to it.

Reply:
> `std::any`

Please consider it to be NO written in the largest font possible. `std::any` does not helps open endianness in any way. How would the NumericConverter knows what is expected by the converter? 

I have not yet seen a place where `any` is an indication of anything but bad design in C++. `void*` can be justified in C due to lack of closures, but not in C++.

This specific place is even worse than EffectSetting: there is no 1-1 correspondence between the user (NumericFormatter) and the implementation (formatter), so such architecture would become unusable. 

I can propose passing a reference to AudacityProject instead, if you are ok with this dependency. This would be a simple and easily extensible solution.

Reply:
Actually, I think that passing AudacityProject would be an only option when we implement the ability to have a different tempo and signature across the project :-)

Reply:
AudacityProject is a sort of all-purpose "context" oftentimes, it's true

Reply:
So should I proceed?

Reply:
I hope this comment proved to be a good stimulus